core/src/main/java/greencity/service/impl/GoalServiceImpl.java

```diff
@@ -34,4 +56,22 @@ public GoalServiceImpl(GoalTranslationRepo goalTranslationRepo, ModelMapper mode
             .map(g -> modelMapper.map(g, GoalDto.class))
             .collect(Collectors.toList());
     }
+
+    @Override
+    public UserGoalResponseDto getUserGoalResponseDto(UserGoal userGoal) {
+        UserGoalResponseDto userGoalResponseDto = userGoalResponseDtoMapper.convert(userGoal);
+        String languageCode = languageService.extractLanguageCodeFromRequest();
+        if (userGoal.getCustomGoal() == null) {
+            Goal goal = goalRepo
+                .findById(userGoal
+                    .getGoal().getId()).orElseThrow(() -> new GoalNotFoundException(GOAL_NOT_FOUND_BY_ID));
+            userGoalResponseDto.setText(goalTranslationRepo.findByGoalAndLanguageCode(goal, languageCode)
+                .orElseThrow(() -> new GoalNotFoundException(GOAL_NOT_FOUND_BY_LANGUAGE_CODE)).getText());
+        } else if (userGoal.getGoal() == null) {
+            CustomGoal customGoal = customGoalRepo.findById(userGoal
+                .getCustomGoal().getId()).orElseThrow(() -> new NotFoundException(CUSTOM_GOAL_NOT_FOUND_BY_ID));
+            userGoalResponseDto.setText(customGoal.getText());
+        }
```

Comment:
Please, move this to a separate private method

Reply:
refactored